### PR TITLE
chore(ci): generate and publish code coverage

### PR DESCRIPTION
**Pull request by dubloom**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@ project(httpd-datadog)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+set(CMAKE_CXX_OUTPUT_EXTENSION_REPLACE ON)
```
Comment (dmehala) [inline CMakeLists.txt:10]: [question] Mind explaining this addition?

File: .github/workflows/dev.yml [hunk 2]
```diff
@@ -55,9 +60,18 @@ jobs:
...
-      # - name: Upload coverage reports to Codecov
-      #   uses: codecov/codecov-action@v4.0.1
-      #   with:
-      #     token: ${{ secrets.CODECOV_TOKEN }}
-      #     slug: DataDog/httpd-datadog
+      - name: Import GNCO files
+        uses: actions/download-artifact@v3
+        with: 
+          name: gcno-artifacts
+          path: build/mod_datadog/CMakeFiles/mod_datadog.dir/src
```
Comment (dmehala) [inline .github/workflows/dev.yml:67]: `.gcno` are metadata on the source code generated at compilation. It's mainly there to helps determining where coverage data should be collected.

On the other hand, `.gcda` holds the actual coverage data from the program's execution.

**Review by dmehala — COMMENTED** [review]
Good start! 🙂

At this stage, the pull request is reporting coverage data based on the metadata used to reconstruct the control flow from profiling data. However, it's missing the actual profiling data - the coverage data itself - which is necessary for the report to be fully useful and actionable.

To make the coverage meaningful, the profiling data should be generated by running the integration tests. This will provide insights into what parts of the code are truly being tested and covered by these tests.

I recommend looking further into _how code coverage works_ to better understand the process.

TL;DR: 
- `.gcno` contains information to reconstruct the control flow.
- `.gcda` contains profiling data collected during a runtime execution of a program.
- Both `.gcno` and `.gcda` are used to generate reports.

File: .github/workflows/dev.yml (hunk 2 above)
Comment (dubloom) [reply .github/workflows/dev.yml:67]: Is the purpose of this comment to say it lacks the .gcda ? Because from what i understood to compute the coverage we need both the .gnco and the .gcda

The .gnco files are generated at compilation as you said so we lose them in the next jobs, that is why i upload them. On the other side, the .gcda we'll be automatically generated in the same folder (as the .gnco files) after the tests. Therefore, we do have the gcda files when I call the lcov command.

File: CMakeLists.txt (hunk 1 above)
Comment (dubloom) [reply CMakeLists.txt]: I had a problem with generating the coverage and it could fix the issue. In fact, it was not the right solution. It is an oversight, i removed it with my new commit

File: .github/workflows/dev.yml (hunk 2 above)
Comment (dmehala) [reply .github/workflows/dev.yml:67]: Indeed. And you had to do all the shenanigans because the module is not built in this step 🤦‍♂️
My apologies than.

**Review by dmehala — APPROVED** [review]
Wrong understanding. You'r good to ship :shipit: 
Can you to add the codecov badge on the README?
